source/auth/Credentials.cpp

```diff
@@ -279,7 +283,9 @@ namespace Aws
                 struct aws_credentials_provider_x509_options raw_config;
                 AWS_ZERO_STRUCT(raw_config);
 
-                raw_config.bootstrap = config.Bootstrap->GetUnderlyingHandle();
+                raw_config.bootstrap =
+                    config.Bootstrap ? config.Bootstrap->GetUnderlyingHandle()
+                                     : ApiHandle::GetOrCreateStaticDefaultClientBootstrap()->GetUnderlyingHandle();
                 raw_config.tls_connection_options = config.TlsOptions.GetUnderlyingHandle();
                 raw_config.thing_name = aws_byte_cursor_from_c_str(config.ThingName.c_str());
                 raw_config.role_alias = aws_byte_cursor_from_c_str(config.RoleAlias.c_str());
```

Comment:
I would like us to completely scrub aws-crt-cpp for all places where EventLoopGroup|HostResolver|ClientBootstrap get passed around, and make it so they ALL work with the new GetOrCreateDefaultXYZ() functions.

So we'd be looking for constructors/creation-functions that take these as args, and for "config/options" classes that have these as members.

Here's me using regex in vscode to search all header files where we might be passing one of these around:
<img width="489" alt="Screen Shot 2022-02-24 at 3 43 03 PM" src="https://user-images.githubusercontent.com/24399397/155626247-b1311de0-2207-436b-a8aa-ce1deafc6f40.png">

Reply:
Should this be a separate PR/ticket? I don't mind making the change, but I'm wondering since, on the surface, it seems like it might be a bigger change.

Reply:
Still need
- ImdsClientConfig::Bootstrap
- CredentialsProviderImdsConfig::Bootstrap
- HttpClientConnectionOptions::Bootstrap